h2/src/main/org/h2/mvstore/tx/Transaction.java

```diff
@@ -120,17 +199,19 @@ public void markStatementEnd() {
      * @param key the key
      * @param oldValue the old value
      */
-    void log(int mapId, Object key, Object oldValue) {
+    void log(int mapId, Object key, VersionedValue oldValue) {
+        long currentState = statusAndLogId.getAndIncrement();
+        long logId = getLogId(currentState);
         store.log(this, logId, mapId, key, oldValue);
```

Comment:
If this call throw an exception `logId` remains incremented. This is not the same as it was before these changes. Is this an expected behavior?

Reply:
You are right it is change in behavior, but why do we care? If exception will be thrown, the whole transaction will be rolled back, and during that process we should tolerate absence of some log entries.